File: PlayWall/src/main/java/de/tobias/playpad/action/actions/cart/handler/PlayPlayHandler.scala
package de.tobias.playpad.action.actions.cart.handler

import de.tobias.playpad.action.InputType
import de.tobias.playpad.action.actions.cart.CartAction
import de.tobias.playpad.pad.{Pad, PadStatus}
import de.tobias.playpad.project.Project
import de.tobias.playpad.viewcontroller.main.IMainViewController

class PlayPlayHandler extends CartActionHandler {
	override def performAction(`type`: InputType,
							   cartAction: CartAction,
							   pad: Pad,
							   project: Project,
							   mainViewController: IMainViewController) = {
		if (`type` eq InputType.PRESSED) {
			pad.setStatus(PadStatus.RESTART);
		}
	}
}

File: PlayWall/src/main/java/de/tobias/playpad/action/actions/cart/handler/PlayPlayHandler.java
package de.tobias.playpad.action.actions.cart.handler;

import de.thecodelabs.midi.event.KeyEventType;
import de.tobias.playpad.action.actions.CartAction;
import de.tobias.playpad.pad.Pad;
import de.tobias.playpad.pad.PadStatus;

public class PlayPlayHandler implements CartActionHandler
{

	@Override
	public void performAction(KeyEventType keyEventType, CartAction cartAction, Pad pad)
	{
		if(keyEventType.equals(KeyEventType.DOWN))
		{
			pad.setStatus(PadStatus.RESTART);
		}
	}
}
